include/event2/ws.h

```diff
@@ -16,14 +16,18 @@ typedef void (*ws_on_msg_cb)(
 typedef void (*ws_on_close_cb)(struct evws_connection *, void *);
 
 /** Opens new WebSocket session from HTTP request.
- *
- * @param req a request object
- * @param cb the callback function that gets invoked on receiving message
- * @param arg an additional context argument for the callback
- * @return a pointer to a newly initialized WebSocket connection or NULL
- *   on error
- * @see evws_close()
- * */
+  @param req a request object
+  @param cb the callback function that gets invoked on receiving message
+  with evbuffer with len bytes of the incoming data available in
+  it. User may use `evbuffer_pullup` to get a pointer to the data, but shouldn't
```

Comment:
Actually I thought about this a little bit more and now it became too unsafe, since user may still call some remove/drain/..., what do you think about passing `const char *` to callback instead of buffer?

Reply:
As for me, first approach (without passed evbuffer to the user) was more clear and user-friendly (if it even could be applied to users of libevent library)

The main issue is that user except `const char *` with terminated \0 for text messages, but I have no idea how to implement this without additional memory allocation and copying

Reply:
Okay, convinced, let's get back to an approach with a separate allocations.
Sorry for the noise.

> The main issue is that user except const char * with terminated \0 for text messages

This is not a an issue, it just should be documented, `const char *` is not always NUL terminated.

Reply:
Fixed